dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Memory.Collections;
+using Microsoft.SemanticKernel.Memory.Storage;
+using Newtonsoft.Json;
+
+namespace Skills.Memory.CosmosDB;
+
+/// <summary>
+/// An implementation of <see cref="IMemoryStore{TEmbedding}"/> for Azure Cosmos DB.
+/// </summary>
+/// <remarks>The Embedding data is saved to the Azure Cosmos DB database container specified in the constructor.
+/// The embedding data persists between subsequent instances and has similarity search capability, handled by the client as Azure Cosmos DB is not a vector-native DB.
+/// </remarks>
+public class CosmosDBMemoryStore<TEmbedding> : IMemoryStore<TEmbedding>, IDisposable
```

Comment:
Is this being used at all yet? Would be nice to have tests.

Reply:
Not yet - tests will be coming shortly.